core/cautils/localgitrepository.go

```diff
@@ -122,82 +111,65 @@ func (g *LocalGitRepository) GetLastCommit() (*apis.Commit, error) {
 	}, nil
 }
 
-func (g *LocalGitRepository) getAllCommits() ([]*git2go.Commit, error) {
-	logItr, itrErr := g.git2GoRepo.Walk()
-	if itrErr != nil {
-
-		return nil, itrErr
+func (g *LocalGitRepository) getAllCommits() ([]*objectv5.Commit, error) {
+	ref, err := g.goGitRepo.Head()
+	if err != nil {
+		return nil, err
 	}
-
-	pushErr := logItr.PushHead()
-	if pushErr != nil {
-		return nil, pushErr
+	logItr, err := g.goGitRepo.Log(&gitv5.LogOptions{From: ref.Hash()})
+	if err != nil {
+		return nil, err
 	}
 
-	var allCommits []*git2go.Commit
-	err := logItr.Iterate(func(commit *git2go.Commit) bool {
+	var allCommits []*objectv5.Commit
+	err = logItr.ForEach(func(commit *objectv5.Commit) error {
 		if commit != nil {
 			allCommits = append(allCommits, commit)
-			return true
 		}
-		return false
+		return nil
 	})
 
 	if err != nil {
 		return nil, err
 	}
 
-	if err != nil {
-		return nil, err
-	}
-
 	return allCommits, nil
 }
 
 func (g *LocalGitRepository) GetFileLastCommit(filePath string) (*apis.Commit, error) {
 	if len(g.fileToLastCommit) == 0 {
 		filePathToCommitTime := map[string]time.Time{}
-		filePathToCommit := map[string]*git2go.Commit{}
+		filePathToCommit := map[string]*objectv5.Commit{}
 		allCommits, _ := g.getAllCommits()
 
 		// builds a map of all files to their last commit
 		for _, commit := range allCommits {
 			// Ignore merge commits (2+ parents)
-			if commit.ParentCount() <= 1 {
+			if commit.NumParents() <= 1 {
 				tree, err := commit.Tree()
 				if err != nil {
 					continue
 				}
 
 				// ParentCount can be either 1 or 0 (initial commit)
 				// In case it's the initial commit, prevTree is nil
-				var prevTree *git2go.Tree
-				if commit.ParentCount() == 1 {
-					prevCommit := commit.Parent(0)
+				var prevTree *objectv5.Tree
+				if commit.NumParents() == 1 {
+					prevCommit, _ := commit.Parent(0)
 					prevTree, err = prevCommit.Tree()
 					if err != nil {
 						continue
 					}
 				}
 
-				diff, err := g.git2GoRepo.DiffTreeToTree(prevTree, tree, nil)
+				changes, err := prevTree.Diff(tree)
```

Comment:
this is the slow part... merkle tree diffs are painfully slow
I have tried to walk around with some `commit.Files()` to skip the tree diffs, but it's still slow to open blobs in order to get files changed.